frontend/src/DeviceDetailsPage/DeviceDetailsPage.jsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Button, Container, Grid } from "@material-ui/core";
+
+import DeviceDetailsConciseTable from "./DeviceDetailsConciseTable";
+import DynamicBreadcrumb from "../general/DynamicBreadcrumb";
+import DeviceDetailsTabTable from "./DeviceDetailsTabTable";
+import DeviceInfo from "../model/DeviceInfo";
+import { getSampleSender } from "../api/SampleData";
+
+export default function DeviceDetailsPage(props) {
+  const {
+    location: {
+      state: { device }
+    }
```

Comment:
because that's how it's passed in the dynamic page routing. I don't make the rules \shrug
https://learnwithparam.com/blog/dynamic-pages-in-react-router/